inc/export/cron/namespace.php

```diff
@@ -0,0 +1,228 @@
+ *
+ * @return void
+ */
+function log( string $message, int $level = E_NOTICE ) : void {
+
+	trigger_error( $message, $level );
+
+	/**
+	 * Trigger custom handling of logs, eg: for external notifications.
+	 *
+	 * @param string $message Log message.
+	 * @param int    $level   Log level, one of PHP error levels.
+	 */
+	do_action( 'altis.analytics.export.log', $message, $level );
+}
+
+/**
+ * Cron job handlers.
+ *
+ * @return void
+ */
+function cron_handler() : void {
+
+	// Check if anyone is listening.
+	$has_handlers = has_action( 'altis.analytics.export.data.process' );
+	if ( ! $has_handlers ) {
+		// No takers ? go home early.
+		return;
+	}
+
+	// Get an S3 client.
+	$client = get_s3_client( [
+		'retries' => [
+			'mode' => 'adaptive',
+			'max_attempts' => 3,
+		],
+	] );
+
+	if ( ! $client ) {
+		return;
+	}
+
+	// Get the files to process.
+	$data = get_analytics_data( $client );
+	if ( ! isset( $data ) ) {
+		return;
+	}
+
+	if ( empty( $data ) ) {
+		log( 'Warning: Empty analytics data were found.', E_WARNING );
+		return;
+	}
+
+	$entries = substr_count( $data, "\n" );
+	log( sprintf( 'Success: Found %d analytics events, delivering payload to subscribers..', $entries ), E_NOTICE );
+
+	/**
+	 * Process analytics data exported from S3.
+	 *
+	 * @param string $data NDJSON data for analytics events since last job run.
+	 */
+	do_action( 'altis.analytics.export.data.process', $data );
+}
+
+/**
+ * Get analytics data from S3 bucket.
+ *
+ * @param S3Client $client S3 client.
+ *
+ * @return string NDJSON string of analytics data.
+ */
+function get_analytics_data( S3Client $client ) : ? string {
+
+	// Check the last processed file key.
+	$last_processed_key = get_option( ALTIS_ANALYTICS_EXPORT_LAST_PROCESSED_KEY );
+
+	/**
+	 * Limit the query to X number of files, in case the last timestamp isn't found.
+	 *
+	 * @param int Number of files to limit the query to.
+	 */
+	$max_files = apply_filters( 'altis.analytics.export.max.files', 100 );
+
+	/**
+	 * Filter the analytics bucket ARN.
+	 *
+	 * @param string Analytics bucket ARN.
+	 */
+	$bucket_arn = apply_filters( 'altis.analytics.export.bucket.arn', ALTIS_ANALYTICS_PINPOINT_BUCKET_ARN );
+
+	// Fetch first batch of items.
+	try {
+		$keys = [];
+
+		$list_params = [
+			'Bucket' => $bucket_arn,
+			'MaxKeys' => $max_files,
+		];
+
+		if ( $last_processed_key ) {
+			$list_params['StartAfter'] = $last_processed_key;
+		}
+
+		$result = $client->listObjectsV2( $list_params );
+
+		// Store keys to delete.
+		foreach ( $result['Contents'] as $item ) {
+			// Check prefix matches a date.
+			if ( ! preg_match( '#^(\d{4}/\d{2}/\d{2})#', $item['Key'], $date_match ) ) {
+				continue;
+			}
+
+			$keys[] = [
+				'Bucket' => $bucket_arn,
+				'Key' => $item['Key'],
+			];
+		}
+
+		// Nothing more to do if there's nothing to delete.
+		if ( empty( $keys ) ) {
+			log( 'Warning: No matching analytics data files were found unlike expected.', E_WARNING );
```

Comment:
I did think about this, added it explicitly because it might be unlikely for a service to use BI services and not expect a whole chunk of time to not have any users. They might need to adjust the frequency of running then, otherwise we'll be burning trees in vain. So it's high likely that some tweaking will need to happen if this starts to show up.